pkg/convert/pprof/streaming/profile_finder.go

```diff
@@ -0,0 +1,117 @@
+package streaming
+
+import (
+	"fmt"
+	"sort"
+)
+
+func newFinder(functions []function, locations []location) finder {
+	res := finder{functions: functions, locations: locations}
+	if !locationSlice(locations) {
+		res.locationsMap = locationMap(locations)
+	}
+	if !functionSlice(functions) {
+		res.functionsMap = functionMap(functions)
+	}
+	return res
+}
+
+type finder struct {
+	functions    []function
+	locations    []location
+	functionsMap map[uint64]*function
+	locationsMap map[uint64]*location
+}
+
+//go:noinline
+func (f *finder) FindLocation(id uint64) (*location, bool) {
+	if f.locationsMap == nil {
+		idx := id - 1
+		//if idx >= uint64(len(f.locations)) {
+		//	return nil, false
+		//}
+		return &f.locations[idx], true
+	}
+	l, ok := f.locationsMap[id]
+	return l, ok
+}
+
+//go:noinline
+func (f *finder) FindFunction(id uint64) (*function, bool) {
+	if f.functionsMap == nil {
+		if id == 0 {
+			fmt.Println("hui")
+		}
+		idx := id - 1
+		//if idx >= uint64(len(f.functions)) {
+		//	return nil, false
+		//}
+		return &f.functions[idx], true
+	}
+	ff, ok := f.functionsMap[id]
+	return ff, ok
+}
```

Comment:
Why not to inline them?

Reply:
Also, please clean up the code from debug helpers

Reply:
noinline is also leftover, will remove